### PR TITLE
Get fragment width to calculate sell size

### PR DESCRIPTION
**Pull request by LumisD**
it's added a function to measure fragment width to calculate correct cell size
Please review

### CODE REVIEW

File: app/src/main/java/com/lumisdinos/gameoffifteen/presentation/HomeViewModel.kt [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+                setFragWidthAndCellSizInConfig(frWidth, gridMargin, cellMargin)
+                //todo: generate cells
+            }
+        }
+    }
+
+
+    private fun setFragWidthAndCellSizInConfig(frWidth: Int, gridMargin: Int, cellMargin: Int) {
+        AppConfig.fragmentWidth = frWidth
+        cell15Size = (frWidth - 2 * gridMargin - 8 * cellMargin) / 4
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/lumisdinos/gameoffifteen/presentation/HomeViewModel.kt]: Hardcode detected. Use Constants

File: app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt [hunk 2]
```diff
@@ -15,6 +19,9 @@ class HomeFragment : DaggerFragment() {
     @Inject
     lateinit var viewModelFactory: ViewModelProvider.Factory
 
+    @Inject
+    lateinit var preference: SharedPreferences
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt]: It's better to use SharedPreferences inside DataRepository

Comment (LumisD) [reply app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt]: as without SharedPreferences in fragment I can't follow the logic - I got rid of it

File: app/src/main/java/com/lumisdinos/gameoffifteen/presentation/HomeViewModel.kt (hunk 1 above)
Comment (LumisD) [reply app/src/main/java/com/lumisdinos/gameoffifteen/presentation/HomeViewModel.kt]: done

File: app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt [hunk 3]
```diff
@@ -25,11 +25,11 @@ class HomeFragment : DaggerFragment() {
...
     ): View? {
         viewBinding = FragmentHomeBinding.inflate(inflater, container, false)
-        val view = viewBinding.root
+        val view = viewBinding?.root
         setHasOptionsMenu(true)
-        viewBinding.root.post {
+        viewBinding?.root?.post {
             viewModel.initialLoadCells(
-                view.width,
+                view!!.width,
```
Comment (artemMprokhorov) [inline app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt]: try don't use not-null assertion , find another solution with safe call or use another construction here

Comment (vladislav-v-rusakov) [reply app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt]: view?.width.or(0) for example

Comment (LumisD) [reply app/src/main/java/com/lumisdinos/gameoffifteen/ui/fragment/HomeFragment.kt]: done

**Review by artemMprokhorov — APPROVED** [review]

**Pull request merged**